src/server/reset_to_version.rs

```diff
@@ -120,57 +120,55 @@ impl ResetToVersionWorker {
         Ok(Batch { writes, has_more })
     }
 
-    pub fn process_next_batch(
-        &mut self,
-        batch_size: usize,
-        wb: &mut RocksWriteBatchVec,
-    ) -> Result<bool> {
-        let Batch { writes, has_more } = self.scan_next_batch(batch_size)?;
+    pub fn process_next_batch(&mut self, wb: &mut RocksWriteBatchVec) -> Result<bool> {
+        let Batch { writes, has_more } = self.scan_next_batch(BATCH_SIZE)?;
         for (key, write) in writes {
             let default_key = Key::from_encoded_slice(&key)
-                .truncate_ts()
-                .unwrap()
+                .truncate_ts()?
                 .append_ts(write.start_ts);
             box_try!(wb.delete_cf(CF_WRITE, &key));
             box_try!(wb.delete_cf(CF_DEFAULT, default_key.as_encoded()));
         }
         if !wb.is_empty() {
-            wb.write().unwrap();
+            wb.write()?;
             wb.clear();
         }
+        // Step into the next step.
+        if !has_more {
+            *self.state.lock().unwrap() = ResetToVersionState::RemovingLock { scanned: 0 };
+        }
         Ok(has_more)
     }
 
-    pub fn process_next_batch_lock(
-        &mut self,
-        batch_size: usize,
-        wb: &mut RocksWriteBatchVec,
-    ) -> Result<bool> {
+    pub fn process_next_batch_lock(&mut self, wb: &mut RocksWriteBatchVec) -> Result<bool> {
         let mut has_more = true;
-        for _ in 0..batch_size {
-            if self.lock_iter.valid().unwrap() {
-                let mut state = self
-                    .state
-                    .lock()
-                    .expect("failed to lock ResetToVersionWorker::state");
+        for _ in 0..BATCH_SIZE {
+            if self.lock_iter.valid()? {
+                let mut state = self.state.lock().unwrap();
                 debug_assert!(matches!(
                     *state,
                     ResetToVersionState::RemovingLock { scanned: _ }
                 ));
                 *state.scanned() += 1;
+                self.total_scanned += 1;
                 drop(state);
 
                 box_try!(wb.delete_cf(CF_LOCK, self.lock_iter.key()));
```

Comment:
User may not pass a `ts` which is larger than the `min-resolved-ts`. What if the primary key is committed, but the secondary lock is left due to error. Better add a paranoid check to make sure the lock's start ts is smaller than provided `ts`.

Reply:
/cc @MyonKeminta